File: only-duplicates/__tests__/onlyDuplicates.test.js
const { onlyDuplicates } = require('../onlyDuplicates');

describe('Given a string, remove all unique characters & return', () => {
    test('Given "abccdefee", return "cceee"', () => {
        const string = "abccdefee";
        expect(onlyDuplicates(string)).toBe("cceee");
    });
    test('Given "hello", return "ll"', () => {
        const string = "hello";
        expect(onlyDuplicates(string)).toBe("ll");
    });
    test('Given "colloquial", return "ollol"', () => {
        const string = "colloquial";
        expect(onlyDuplicates(string)).toBe("ollol");
    });
    test('Given "foundersandcoders", return "ondersndoders"', () => {
        const string = "foundersandcoders";
        expect(onlyDuplicates(string)).toBe("ondersndoders");
    });
    test('Given "ehfgfffaihf", return "hffffhf"', () => {
        const string = "ehfgfffaihf";
        expect(onlyDuplicates(string)).toBe("hffffhf");
    });
    test('Given "hhhgfgedeei", return "hhhggeee"', () => {
        const string = "hhhgfgedeei";
        expect(onlyDuplicates(string)).toBe("hhhggeee");
    });
    test('Given "eehgfieeecg", return "eegeeeg"', () => {
        const string = "eehgfieeecg";
        expect(onlyDuplicates(string)).toBe("eegeeeg");
    });
    test('Given "egbhceceifh", return "ehceceh"', () => {
        const string = "egbhceceifh";
        expect(onlyDuplicates(string)).toBe("ehceceh");
    });
    test('Given "ifegfehhfhc", return "fefehhfh"', () => {
        const string = "ifegfehhfhc";
        expect(onlyDuplicates(string)).toBe("fefehhfh");
    });
});

//https://www.codewars.com/kata/5a1dc4baffe75f270200006b/train/javascript

File: python-venv/only_duplicates/__tests__/only_duplicates_test.py
from kata_solution.only_duplicates import only_duplicates

def test_1():
    string = "abccdefee"
    assert only_duplicates(string) == "cceee"

def test_2():
    string = "hello"
    assert only_duplicates(string) == "ll"

def test_3():
    string = "colloquial"
    assert only_duplicates(string) == "ollol"

def test_4():
    string = "foundersandcoders"
    assert only_duplicates(string) == "ondersndoders"

def test_5():
    string = "ehfgfffaihf"
    assert only_duplicates(string) == "hffffhf"

def test_6():
    string = "hhhgfgedeei"
    assert only_duplicates(string) == "hhhggeee"

def test_7():
    string = "eehgfieeecg"
    assert only_duplicates(string) == "eegeeeg"

def test_8():
    string = "egbhceceifh"
    assert only_duplicates(string) == "ehceceh"

def test_9():
    string = "ifegfehhfhc"
    assert only_duplicates(string) == "fefehhfh"



#https://www.codewars.com/kata/5a1dc4baffe75f270200006b/train/python